Listenbrainz/Repository/Feed/FeedRepositoryImpl.swift

```diff
@@ -22,7 +21,123 @@ class FeedRepositoryImpl: FeedRepository {
             .validate()
             .publishDecodable(type: FeedAlbum.self)
             .value()
+            .mapError { $0.asAFError ?? .explicitlyCancelled }
+            .eraseToAnyPublisher()
+    }
+
+    func fetchCoverArt(url: URL) -> AnyPublisher<Data, AFError> {
+        return AF.request(url, method: .get)
+            .validate()
+            .publishData()
+            .value()
+            .mapError { $0.asAFError ?? .explicitlyCancelled }
+            .eraseToAnyPublisher()
+    }
+
+    func pinTrack(recordingMsid: String, recordingMbid: String?, blurbContent: String?, userToken: String) -> AnyPublisher<Void, AFError> {
+        let url = URL(string: "\(BuildConfiguration.shared.API_LISTENBRAINZ_BASE_URL)/pin")!
+        let parameters: [String: Any] = [
+            "recording_msid": recordingMsid,
+            "recording_mbid": recordingMbid ?? NSNull(),
+            "blurb_content": blurbContent ?? "",
+            "pinned_until": Int(Date().timeIntervalSince1970 + 7 * 24 * 60 * 60) // 1 week in the future
```

Comment:
What, and why?

Reply:
discussed on IRC, do you want me to keep the logic same?

Reply:
Works